powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/CallbackUrlBehavior.java

```diff
@@ -416,10 +423,10 @@ private void notifyCallbackUrl(CallbackUrlEntity callbackUrlEntity, Map<String,
      * @throws RestClientException Thrown when rest client initialization fails.
      */
     private synchronized RestClient getRestClient(CallbackUrlEntity callbackUrlEntity) throws RestClientException {
-        RestClient restClient = restClientCache.get(callbackUrlEntity);
+        RestClient restClient = restClientCache.get(callbackUrlEntity.getId());
         if (restClient == null) {
             restClient = initializeRestClient(callbackUrlEntity);
-            restClientCache.put(callbackUrlEntity, restClient);
+            restClientCache.put(callbackUrlEntity.getId(), restClient);
```

Comment:
To be super safe for further refactoring, we may introduce a method to retrieve a cache key and encapsulate `callbackUrlEntity.getId()`.

Reply:
Sure, we can do this.